deployments/azure-webapps-container.yml

```diff
@@ -0,0 +1,75 @@
+# This workflow will build and push a Docker container to an Azure Web App when a commit is pushed to your default branch.
+#
+# This workflow assumes you have already created the target Azure App Service web app.
+# For instructions see https://docs.microsoft.com/en-us/azure/app-service/quickstart-custom-container?tabs=dotnet&pivots=container-linux
+#
+# To configure this workflow:
+#
+# 1. Download the Publish Profile for your Azure Web App. You can download this file from the Overview page of your Web App in the Azure Portal.
+#    For more information: https://docs.microsoft.com/en-us/azure/app-service/deploy-github-actions?tabs=applevel#generate-deployment-credentials
+#
+# 2. Create a secret in your repository named AZURE_WEBAPP_PUBLISH_PROFILE, paste the publish profile contents as the value of the secret.
+#    For instructions on obtaining the publish profile see: https://docs.microsoft.com/azure/app-service/deploy-github-actions#configure-the-github-secret
+#
+# 3. Change the value for the AZURE_WEBAPP_NAME.
+#
+# For more information on GitHub Actions for Azure: https://github.com/Azure/Actions
+# For more information on the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+# For more samples to get started with GitHub Action workflows to deploy to Azure: https://github.com/Azure/actions-workflow-samples
+
+name: Build and deploy a container to an Azure Web App
+
+env:
+  AZURE_WEBAPP_NAME: your-app-name  # set this to the name of your Azure Web App
+
+on:
+  push:
+    branches:
+      - $default-branch
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to GitHub container registry
+        uses: docker/login-action@v1.10.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Can you also list the [permissions](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idpermissions) that are required for GITHUB_TOKEN for this job?

Reply:
@JasonFreeberg you can refer to this [doc](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow) to find what permission to specify

Reply:
@skedwards88 -- I was out sick, thanks for your patience. The required permissions are "repo" and "read:packages". I updated the workflow to use the default workflow token (`github.token`), but the PAT should still be created and provided to the Azure web app. I'll provide more details on the docs PR

Reply:
@JasonFreeberg IIUC, `github.token` and `secrets.GITHU_TOKEN` are functionally equivalent, and permissions should be specified for both. I think you will want to add the `permissions` key to this workflow; otherwise, the workflow will fail for users who grant GITHUB_TOKEN limited scope by default.

Reply:
@JasonFreeberg GITHUB_TOKEN permissions are different from PAT scopes. I am guessing that the permissions are ` contents: read` and `packages: write`, but confirmation before the corresponding guide for this template will be helpful: https://github.com/github/docs-internal/pull/22832